lib/oneview-sdk/resource.rb

```diff
@@ -361,6 +361,12 @@ def recursive_like?(other, data = @data)
         return false unless data && data.respond_to?(:[])
         if val.is_a?(Hash)
           return false unless data.class == Hash && recursive_like?(val, data[key.to_s])
+        elsif val.is_a?(Array) && val.first.is_a?(Hash)
```

Comment:
Array of Hashes in OneView does not require any specific order, actually, if you look close into the hashes in these types of arrays, each has an index or sort of defined inside. (e.g. Server Profile connections have connectionID that is a index, or in Logical Interconnect Groups  interconnectMapTemplates that each specify bay and enclosure index)